eureka-client/src/main/java/com/netflix/discovery/converters/StringCache.java

```diff
@@ -0,0 +1,38 @@
+package com.netflix.discovery.converters;
+
+import java.util.Map;
+import java.util.WeakHashMap;
+
+/**
+* @author Tomasz Bak
+*/
+public class StringCache {
```

Comment:
Using String.intern() was technically forbidden in the past for string pool usage. This implementation of StringCache follows xstream's StringConverter which I tried to use, but since we have custom serializer it is never invoked.
I checked again, and actually in Java7 the String.intern() implementation is fixed, and it is better alternative than having own string pools, but unless running on 7u40, it requires setting -XX:StringTableSize=N to proper value (before 7u40 default = 1009, later default = 60013 which is better but we might need even higher value). Check this article: http://java-performance.info/string-intern-in-java-6-7-8/
Given that we do not control which Java7 JVM versions clients run, and we cannot enforce proper JVM configuration, sticking to own String pool implementation seems to be better option short term. Medium/long term, if we want to make similar optimization in Eureka 2.x I would go for String.intern.